pandas/tests/io/formats/test_to_html.py

```diff
@@ -22,6 +23,113 @@
     pass
 
 
+def _has_names(index):
+    if index.nlevels > 1:
+        return com._any_not_none(*index.names)
+    else:
+        return index.name is not None
+
+
+def _index_init_params(index):
+    return (index.nlevels > 1,
+            _has_names(index))
+
+
+@pytest.fixture
```

Comment:
Is this going to be used outside this module? If not then can probably just be a global function instead of a fixture. If so then I think would make more sense to move fixtures to a conftest for this package

Reply:
It will not be of use outside this module. It uses `read_file` which uses the `datapath` fixture. So does it not need to be a fixture or am I doing something wrong?

Reply:
I think it would be a good idea if the methodology used here, if approved, is then used to move all the inline HTML from the other tests into separate files. Would be easier to check the expected output of other tests. Would be easier to re-use the expect outputs of other tests. And would allow much larger outputs/comprehensive tests to be used in future tests without cluttering the test file.

I'll move it all into one fixture for the moment. It can always be separated out again in the future.

Reply:
In  #22655, I've added a issue specific test as well as the parametric regression tests. So I am already using `expected_html` in more than one test and will need to leave it as it is.